core/src/test/java/org/apache/iceberg/TestBaseIncrementalAppendScan.java

```diff
@@ -50,6 +50,42 @@ public void testFromSnapshotInclusive() {
     Assert.assertEquals(3, Iterables.size(scanWithToSnapshot.planFiles()));
   }
 
+  @Test
+  public void testFromSnapshotInclusiveWithBranch() {
```

Comment:
it would also be good to add tests that hit the `Cannot find the starting snapshot` / `Cannot find ref...` error messages